aten/src/ATen/native/cuda/Gesv.cu

```diff
@@ -138,6 +166,76 @@ std::tuple<Tensor,Tensor> _gesv_helper_cuda(const Tensor& self, const Tensor& A)
   return std::tuple<Tensor,Tensor>(b_working_copy, A_working_copy);
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cuda(Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_MAGMA
+AT_ERROR("gesv: MAGMA library not found in "
+    "compilation. Please rebuild with MAGMA.");
+#else
+  /* See Gesv.cpp for comments */
```

Comment:
A lot of this code is duplicated between gesv.cpp and gesv.cu. Could we refactor out the common bits?

As I understand it most of this code is coercing the inputs into column major format, right?